src/app/product-detail/product-detail.component.ts

```diff
@@ -0,0 +1,56 @@
+import { Component, OnInit } from '@angular/core';
+import {Product} from "../model/product";
+import {ActivatedRoute} from "@angular/router";
+import {ProductsService} from "../services/products.service";
+import { Location } from '@angular/common';
+import {OrderService} from "../services/order.service";
+import {LoginService} from "../services/login.service";
+
+@Component({
+  selector: 'app-product-detail',
+  templateUrl: './product-detail.component.html',
+  styleUrls: ['./product-detail.component.scss']
+})
+export class ProductDetailComponent implements OnInit {
+
+  constructor(private route: ActivatedRoute, private productService: ProductsService, private location: Location, private orderService: OrderService, private  loginService: LoginService) { }
+
+  product : Product | undefined;
+
+  isAdmin = this.loginService.isAdmin();
```

Comment:
**suggestion:** you should have this assignment in the component lifecycle, more specifically in `ngOnInit`